Content.Server/Chemistry/EntitySystems/ChemistrySystemHypospray.cs

```diff
@@ -30,23 +38,101 @@ private void OnSolutionChange(EntityUid uid, HyposprayComponent component, Solut
+            TryDoInject(component, args.HitEntities.First(), args.User);
+        }
+
+        public bool TryDoInject(HyposprayComponent component, EntityUid? target, EntityUid user)
+        {
+            if (!EligibleEntity(target, _entMan))
+                return false;
+
+            string? msgFormat = null;
+
+            if (target == user)
+            {
+                msgFormat = "hypospray-component-inject-self-message";
+            }
+            else if (EligibleEntity(user, _entMan) && _interaction.TryRollClumsy(user, component.ClumsyFailChance))
+            {
+                msgFormat = "hypospray-component-inject-self-clumsy-message";
+                target = user;
+            }
+
+            _solutions.TryGetSolution(component.Owner, component.SolutionName, out var hypoSpraySolution);
+
+            if (hypoSpraySolution == null || hypoSpraySolution.CurrentVolume == 0)
+            {
+                _popup.PopupCursor(Loc.GetString("hypospray-component-empty-message"), Filter.Entities(user));
+                return true;
+            }
+
+            if (!_solutions.TryGetInjectableSolution(target.Value, out var targetSolution))
+            {
+                _popup.PopupCursor(Loc.GetString("hypospray-cant-inject", ("target", Identity.Entity(target.Value, _entMan))), Filter.Entities(user));
+                return false;
+            }
+
+            _popup.PopupCursor(Loc.GetString(msgFormat ?? "hypospray-component-inject-other-message", ("other", target)), Filter.Entities(user));
+
+            if (target != user)
+            {
+                _popup.PopupCursor(Loc.GetString("hypospray-component-feel-prick-message"), Filter.Entities(target.Value));
+                var meleeSys = EntitySystem.Get<MeleeWeaponSystem>();
+                var angle = Angle.FromWorldVec(_entMan.GetComponent<TransformComponent>(target.Value).WorldPosition - _entMan.GetComponent<TransformComponent>(user).WorldPosition);
+                // TODO: This should just be using melee attacks...
+                // meleeSys.SendLunge(angle, user);
+            }
+
+            _audio.PlayPvs(component.InjectSound, user);
+
+            // Get transfer amount. May be smaller than component.TransferAmount if not enough room
+            var realTransferAmount = FixedPoint2.Min(component.TransferAmount, targetSolution.AvailableVolume);
+
+            if (realTransferAmount <= 0)
+            {
+                _popup.PopupCursor(Loc.GetString("hypospray-component-transfer-already-full-message",("owner", target)), Filter.Entities(user));
+                return true;
+            }
+
+            // Move units from attackSolution to targetSolution
+            var removedSolution = _solutions.SplitSolution(component.Owner, hypoSpraySolution, realTransferAmount);
+
+            if (!targetSolution.CanAddSolution(removedSolution))
+            {
+                return true;
+            }
```

Comment:
```suggestion
            if (!targetSolution.CanAddSolution(removedSolution))
                return true;
```

Also here for consistency too.